tests/phpunit/tests/option/networkOption.php

```diff
@@ -123,86 +123,52 @@ public function data_network_id_parameter() {
 	}
 
 	/**
-	 * @ticket 43506
+	 * @ticket 37181
 	 * @group ms-required
 	 */
-	public function test_get_network_option_sets_notoptions_if_option_found() {
-		$network_id     = get_current_network_id();
-		$notoptions_key = "$network_id:notoptions";
+	public function test_meta_api_use_values_in_network_option() {
+		$network_id = self::factory()->network->create();
+		$option     = __FUNCTION__;
+		$value      = __FUNCTION__;
 
-		$original_cache = wp_cache_get( $notoptions_key, 'site-options' );
-		if ( false !== $original_cache ) {
-			wp_cache_delete( $notoptions_key, 'site-options' );
-		}
-
-		// Retrieve any existing option.
-		get_network_option( $network_id, 'site_name' );
-
-		$cache = wp_cache_get( $notoptions_key, 'site-options' );
-		if ( false !== $original_cache ) {
-			wp_cache_set( $notoptions_key, $original_cache, 'site-options' );
-		}
-
-		$this->assertSame( array(), $cache );
+		add_metadata( 'site', $network_id, $option, $value, true );
+		$this->assertEquals( get_metadata( 'site', $network_id, $option ), array( get_network_option( $network_id, $option, true ) ) );
 	}
 
 	/**
-	 * @ticket 43506
+	 * @ticket 37181
 	 * @group ms-required
 	 */
-	public function test_get_network_option_sets_notoptions_if_option_not_found() {
-		$network_id     = get_current_network_id();
-		$notoptions_key = "$network_id:notoptions";
-
-		$original_cache = wp_cache_get( $notoptions_key, 'site-options' );
-		if ( false !== $original_cache ) {
-			wp_cache_delete( $notoptions_key, 'site-options' );
-		}
-
-		// Retrieve any non-existing option.
-		get_network_option( $network_id, 'this_does_not_exist' );
-
-		$cache = wp_cache_get( $notoptions_key, 'site-options' );
-		if ( false !== $original_cache ) {
-			wp_cache_set( $notoptions_key, $original_cache, 'site-options' );
-		}
-
-		$this->assertSame( array( 'this_does_not_exist' => true ), $cache );
```

Comment:
I'm guessing it can safely be removed because the meta version of this implementation already has its own tests?

Is there a sane way (does it _possibly_ make sense) to keep this test around but modify its internals, just for the sake of procedural code coverage?

Reply:
notoptions cache is not needed in this implementation. It is removed, so the test should be removed.

Reply:
I understand. 😅

But... what happens (with this PR applied) when `get_network_option()` is called multiple times using an option key that is known not to exist in the database?

Currently, that miss is cached using the `notoptions` cache key to prevent multiple database misses from happening.

Is that still the case, and should it continue to be tested? (I think, probably?)